src/gauging.jl

```diff
@@ -73,98 +78,66 @@ function vidal_gauge(
-  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+  cache![] = update(cache![]; cache_update_kwargs...)
+
+  return vidal_gauge(ψ, bond_tensors, cache![]; kwargs...)
+end
 
-  mts = belief_propagation(pψψ; niters, target_precision=target_canonicalness, verbose)
-  return vidal_gauge(
-    ψ, pψψ, mts; eigen_message_tensor_cutoff, regularization, svd_kwargs...
-  )
+"""Put an ITensorNetwork into the vidal gauge (by computing the message tensors), return the network and the bond tensors."""
+function vidal_gauge(ψ::ITensorNetwork; kwargs...)
+  bond_tensors = initialize_bond_tensors(ψ)
+  return vidal_gauge(ψ, bond_tensors; kwargs...)
 end
 
 """Transform from an ITensor in the Vidal Gauge (bond tensors) to the Symmetric Gauge (partitionedgraph, message tensors)"""
 function vidal_to_symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
   ψsymm = copy(ψ)
   ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
-  ψsymm_mts = default_bp_cache(pψψsymm)
+
+  bp_cache = BeliefPropagationCache(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
+  mts = messages(bp_cache)
 
   for e in edges(ψsymm)
     vsrc, vdst = src(e), dst(e)
-    pe = partitionedge(pψψsymm, NamedEdge((vsrc, 1) => (vdst, 1)))
+    pe = partitionedge(bp_cache, (vsrc, 1) => (vdst, 1))
     root_S = sqrt_diag(bond_tensors[e])
     setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vsrc]), vsrc)
     setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vdst]), vdst)
 
-    ψsymm_mts[pe], ψsymm_mts[reverse(pe)] = copy(ITensor[dense(bond_tensors[e])]),
-    copy(ITensor[dense(bond_tensors[e])])
+    haskey(mts, pe) && delete!(mts, pe)
+    haskey(mts, reverse(pe)) && delete!(mts, reverse(pe))
+    insert!(mts, pe, copy(ITensor[dense(bond_tensors[e])]))
+    insert!(mts, reverse(pe), copy(ITensor[dense(bond_tensors[e])]))
   end
 
   ψψsymm = norm_network(ψsymm)
   pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
 
-  return ψsymm, pψψsymm, ψsymm_mts
+  return ψsymm, BeliefPropagationCache(pψψsymm, mts, default_message(bp_cache))
 end
 
 """Put an ITensorNetwork into the symmetric gauge and also return the message tensors (which are the diagonal bond matrices from the Vidal Gauge)"""
-function symmetric_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  svd_kwargs...,
-)
-  ψ_vidal, bond_tensors = vidal_gauge(
-    ψ;
-    eigen_message_tensor_cutoff,
-    regularization,
-    niters,
-    target_canonicalness,
-    svd_kwargs...,
-  )
+function symmetric_gauge(ψ::ITensorNetwork; kwargs...)
+  ψ_vidal, bond_tensors = vidal_gauge(ψ; kwargs...)
 
   return vidal_to_symmetric_gauge(ψ_vidal, bond_tensors)
 end
 
 """Transform from the Symmetric Gauge (message tensors) to the Vidal Gauge (bond tensors)"""
 function symmetric_to_vidal_gauge(
```

Comment:
Is this function needed, or can we just use `vidal_gauge`?

Reply:
I don't quite understand your comment, since my comment was on the function called `symmetric_to_vidal_gauge`, but you say:

> it puts an ITensorNetwork into the symmetric gauge

but it looks like this was changed to `vidal_gauge` by now, if I understand the code diffs correctly.